extensions/amp-access-fewcents/0.1/fewcents-impl.js

```diff
@@ -0,0 +1,184 @@
+
+    // Install styles.
+    installStylesForDoc(this.ampdoc, CSS, () => {}, false, TAG);
+  }
+
+  /**
+   * Decides to show the paywall or publisher content by calling authorize endpoint
+   * @return {!Promise<!JsonObject>}
+   */
+  authorize() {
+    return this.getPaywallData_().then(
+      // [TODO] : More edge cases to be added with actual call
+      (response) => {
+        // Removing the paywall from dialogContainer
+        this.emptyContainer_();
+        // Showing the publisher's content
+        return {access: response.data.access};
+      },
+      (err) => {
+        // Rendering the paywall when request promise is rejected
+        const {response} = err;
+        return response.json().then(() => {
+          this.emptyContainer_().then(this.renderPurchaseOverlay_.bind(this));
+          return {access: false};
+        });
+      }
+    );
+  }
+
+  /**
+   * Add request parameters for the authorize endpoint
+   * @return {string}
+   * @private
+   */
+  prepareAuthorizeUrl_() {
+    // [TODO]: Actual implementation
+    dev().fine(TAG, 'Publishers config', this.fewcentsConfig_);
+    return;
+  }
+
+  /**
+   * Get paywall data by calling authorize endpoint
+   * @return {!Promise<Object>}
+   * @private
+   */
+  getPaywallData_() {
+    // [TODO]: Actual API implementation using fetch
+    dev().fine(TAG, 'authorizeUrl', this.authorizeUrl_, CONFIG_BASE_PATH);
+
+    // Currently, case only for showing the paywall later in future depend upon API response
+    return Promise.reject({
+      response: {
+        status: 402,
+        json() {
+          return Promise.resolve({success: true});
+        },
+      },
+    });
+  }
+
+  /**
+   * Removes the paywall from the element on publisher's page where paywall is displayed
+   * @private
+   * @return {!Promise}
+   */
+  emptyContainer_() {
+    // [TODO]: Actual implementation
+    return Promise.resolve();
+  }
+
+  /**
+   * Creates new element for the paywall
+   * @param {string} name
+   * @return {!Element}
+   * @private
+   */
+  createElement_(name) {
+    return createElementWithAttributes(this.ampdoc.win.document, name, {});
+  }
+
+  /**
+   * Return element on publisher's page where paywall will be displayed
+   * @return {!Element}
+   * @private
+   */
+  getPaywallContainer_() {
+    const containerId = this.fewcentsConfig_['contentSelector'];
+    const dialogContainer = this.ampdoc.getElementById(containerId);
+    return user().assertElement(
+      dialogContainer,
+      'No element found with given id',
+      containerId
+    );
+  }
+
+  /**
+   * Creates the paywall component and append it to dialog container
+   * @private
+   */
+  renderPurchaseOverlay_() {
+    // [TODO]: Create entire paywall element with button event listener for login flow
+    this.dialogContainer_ = this.getPaywallContainer_();
+    this.innerContainer_ = this.createElement_('div');
+    this.innerContainer_.className = TAG_SHORTHAND + '-container';
```

Comment:
this code might be cleaner if you just use `createElementWithAttributes(ampdoc, 'div', {className: foo})`

Reply:
or pass attributes in your helper method.

Reply:
Changes done.